src/test/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesQueueTaskDispatcherTest.java

```diff
@@ -73,6 +74,36 @@ public void setUpTwoClouds() throws Exception {
         slaveB = new KubernetesSlave("B", new PodTemplate(), "testB", "B", "dockerB", new KubernetesLauncher(), RetentionStrategy.INSTANCE);
     }
 
+    public void setUpTwoCloudsPermittedCloudsForm() throws Exception {
+        folderA = new Folder(jenkins.jenkins, "A");
+        folderB = new Folder(jenkins.jenkins, "B");
+        jenkins.jenkins.add(folderA, "Folder A");
+        jenkins.jenkins.add(folderB, "Folder B");
+
+        KubernetesCloud cloudA = new KubernetesCloud("A");
+        cloudA.setUsageRestricted(true);
+
+        KubernetesCloud cloudB = new KubernetesCloud("B");
+        cloudB.setUsageRestricted(true);
+
+        jenkins.jenkins.clouds.add(cloudA);
+        jenkins.jenkins.clouds.add(cloudB);
+
+        KubernetesFolderProperty property1 = new KubernetesFolderProperty();
+        folderA.addProperty(property1);
+        JSONObject json1 = new JSONObject();
+        json1.element("permittedClouds", Collections.singletonList("A"));
+        folderA.addProperty(property1.reconfigure(null, json1));
+
+        KubernetesFolderProperty property2 = new KubernetesFolderProperty();
+        folderB.addProperty(property2);
+        JSONObject json2 = new JSONObject();
+        json2.element("permittedClouds", Collections.singletonList("B"));
+        folderB.addProperty(property2.reconfigure(null, json2));
```

Comment:
This sort of test is helpful to verify modifications to the configuration (you can also test more realistically using `WebClient` though it may be trickier to screen-scrape). Is there existing test coverage for this folder property using `configRoundtrip` to verify that if you started with some list of permitted clouds and then opened the **Configure** page of the folder and immediately **Save**’d that you still have the same list?

Reply:
Will add that, in the meantime, already checked config doesn't change:
Pre-change
```xml 
<?xml version='1.1' encoding='UTF-8'?>
<com.cloudbees.hudson.plugins.folder.Folder plugin="cloudbees-folder@6.714.v79e858ef76a_2">
  <actions/>
  <description></description>
  <properties>
[...]
    <org.csanchez.jenkins.plugins.kubernetes.KubernetesFolderProperty plugin="kubernetes@3580.v78271e5631dc">
      <permittedClouds>
        <string>kubernetes</string>
      </permittedClouds>
    </org.csanchez.jenkins.plugins.kubernetes.KubernetesFolderProperty>
    <com.cloudbees.hudson.plugins.folder.properties.SubItemFilterProperty plugin="cloudbees-folders-plus@3.26"/>
  </properties>
[...]
<icon class="com.cloudbees.hudson.plugins.folder.icons.StockFolderIcon"/>
</com.cloudbees.hudson.plugins.folder.Folder>%
```
Post change:
```xml
<?xml version='1.1' encoding='UTF-8'?>
<com.cloudbees.hudson.plugins.folder.Folder plugin="cloudbees-folder@6.714.v79e858ef76a_2">
  <properties>
[...]
    <org.csanchez.jenkins.plugins.kubernetes.KubernetesFolderProperty plugin="kubernetes@999999-SNAPSHOT">
      <permittedClouds>
        <string>kubernetes</string>
      </permittedClouds>
    </org.csanchez.jenkins.plugins.kubernetes.KubernetesFolderProperty>
  </properties>
[...]
  <icon class="com.cloudbees.hudson.plugins.folder.icons.StockFolderIcon"/>
</com.cloudbees.hudson.plugins.folder.Folder>%
```

Reply:
Added a test `KubernetesFolderPropertyTest` with `configRoundTrip` and removed changes added in `KubernetesQueueTaskDispatcherTest` to test new form format, old format is still covered there to guarantee backwards compatibility